components/src/widget/views/interaction/interaction.ts

```diff
@@ -108,26 +117,104 @@ export class PaymentInteraction extends LitElement {
       })
 
       if (!response.ok) {
-        this.currentView = 'failed'
-        this.errorMessage = 'Failed to process payment. Please try again.'
-        this.requestUpdate()
+        this.handleInteractionFail(
+          'Failed to process payment. Please try again'
+        )
         return
       }
 
       const result = (await response.json()) as CheckPaymentResult
+
       if (result.success === false) {
-        this.currentView = 'failed'
-        this.errorMessage = result.error.message
-        this.requestUpdate()
+        this.handleInteractionFail(
+          result.error?.message || 'Payment processing failed'
+        )
         return
       }
 
       this.currentView = 'success'
       this.requestUpdate()
     } catch {
-      this.currentView = 'failed'
-      this.errorMessage = 'There was an issues with your request.'
-      this.requestUpdate()
+      this.handleInteractionFail('There was an issue with your request')
+    }
+  }
+
+  private handleInteractionFail(message: string) {
+    this.currentView = 'failed'
+    this.errorMessage = message
+    this.requestUpdate()
+  }
+
+  private async _startLongPolling({
+    maxAttempts = 3,
+    signal = this._pollingAbortController?.signal
+  }: Partial<{
+    maxAttempts?: number
+    signal?: AbortSignal
+  }> = {}): Promise<void> {
+    if (this._interactionCompleted) return
+    let attempt = 0
+
+    while (++attempt <= maxAttempts) {
+      signal?.throwIfAborted()
+
+      try {
+        const { paymentId } = this.configController.state
+        const { apiUrl } = this.configController.config
+        const url = new URL(`/payment/status/${paymentId}`, apiUrl).href
+
+        const res = await fetch(url, { signal })
+
+        if (res.ok) {
+          const data = (await res.json()) as PaymentStatus
+          switch (true) {
```

Comment:
While I'm personally a fan of `switch (true)`, other people reading the code might get confused. Let's stick to regular if-else.